### ISSUE TITLE
Implement filtering in reference implementation

### ISSUE THREAD

**Issue by R2wenD2** [grafeas/grafeas#29]

**Comment by hpejustin** [comment]
i think a go-restful based framework should be imported, https://github.com/hpejustin/depark this can be a good example.

**Comment by hpejustin** [comment]
i think, we can add Label or Annotation properties to Note and Occurrence, in that case we can use filter to list them.

**Comment by R2wenD2** [comment]
The intention here is to allow filtering on fields that already exist as a part of the occurrence or note. I don't see a case for additional labels or annotations.

The implementation will to parse a query string, reading key,values where keys are the fields of our resources.

**Comment by R2wenD2** [comment]
We'll use one of the following filtering specs:

- https://github.com/google/cel-spec
- https://cloud.google.com/logging/docs/view/advanced_filters

Both of them require an accessible parser - I'm following up on that.

**Comment by furuholm** [comment]
One aspect of this could be to go for a format that can be transformed into a database query, or will that be too restrictive?

**Comment by R2wenD2** [comment]
We need a few things:
1) A parser that will parse and validate a filter expression from query string
2) A mapping of Keys to database fields (which I think is what you're referring to, @furuholm 

I'm working with the cel team to create part one of this, but once we have that part 2 can follow quickly.

We also need a real database, but that's a separate issue.

**Comment by furuholm** [comment]
Added #84 to address the database

**Comment by TristonianJones** [comment]
Hey there, I've implemented the filter parsing using the syntax of [advanced filters](https://cloud.google.com/logging/docs/view/advanced_filters) with the backing representation as [cel](https://github.com/google/cel-spec). Any preference on which directory the parser should go into?

**Comment by R2wenD2** [comment]
@TristonianJones Awesome! Lets put it in grafeas/server-go/filtering.

**Comment by TristonianJones** [comment]
Just curious, but any thoughts on whether this is something that you'd like to continue using? If so, the github.com/google/cel-go library can be used to evaluate ASTs generated by the filtering parser.

**Comment by vtsao** [comment]
Yes we are still planning on using what you impl'd for our filtering. Also thanks for the code pointer. @aysylu FYI this might be something you will take on.
